rsmtool/utils/commandline.py

```diff
@@ -712,37 +712,37 @@ def _finalize(self, user_input):
 
         Parameters
         ----------
-        user_input : list or str
-            Description
+        user_input : Union[List[str], str]
+            The user input for the field that is to be converted.
 
         Returns
         -------
-        final value
+        final value: Any
             The converted value.
         """
         if (user_input == "" or user_input == []) and self.field_type == "optional":
             final_value = DEFAULTS.get(self.field_name)
         else:
             # boolean fields need to be converted to actual booleans
-            if self.data_type == "boolean":
+            if isinstance(user_input, str) and self.data_type == "boolean":
                 final_value = False if user_input == "false" else True
             # and integer fields to integers/None
-            elif self.data_type == "integer":
+            elif isinstance(user_input, str) and self.data_type == "integer":
                 final_value = int(user_input)
             else:
                 final_value = user_input
 
         return final_value
 
-    def get_value(self):
+    def get_value(self) -> Union[bool, int, str, List[str]]:
         """
         Get value of instantiated interactive field.
 
         This is the main public method for this class.
 
         Returns
         -------
-        final_value : list or str
+        final_value : Union[bool, int, str, List[str]]
             The final value of the field which may be a list of
             strings or a string.
```

Comment:
```suggestion
            strings or a string or a bool or an int.
```
I know you'll rephrase it nicely